src/containers/User/Login.tsx

```diff
@@ -1,66 +1,135 @@
-import React, { useState } from 'react'
+import React, { useState, useEffect } from 'react'
 import { useRouter } from '../../utils/hooks/useRouter'
 import { useUserState } from '../../contexts/UserState'
 import { Link } from 'react-router-dom'
-import { Form, Button, Container, Grid, Segment, Header } from 'semantic-ui-react'
+import { Form, Button, Container, Grid, Segment, Header, Dropdown } from 'semantic-ui-react'
 import config from '../../config.json'
 import isLoggedIn from '../../utils/helpers/loginCheck'
 import strings from '../../utils/constants'
-import setUserInfo from '../../utils/helpers/fetchUserInfo'
+import { postRequest as attemptLogin } from '../../utils/helpers/fetchMethods'
+import { User, LoginPayload, OrganisationSimple, TemplatePermissions } from '../../utils/types'
+
+const loginURL = config.serverREST + '/login'
+const loginOrgURL = config.serverREST + '/login-org'
 
 const Login: React.FC = () => {
   const [username, setUsername] = useState('')
   const [password, setPassword] = useState('')
   const [isError, setIsError] = useState(false)
+  const [user, setUser] = useState<User>()
+  const [orgList, setOrgList] = useState<OrganisationSimple[]>([])
+  const [JWT, setJWT] = useState<string>()
+  const [templatePermissions, setTemplatePermissions] = useState<TemplatePermissions>()
+  const [selectedOrgIndex, setSelectedOrgIndex] = useState(0)
   const { push, history } = useRouter()
-  const { login } = useUserState()
+  const { onLogin } = useUserState()
+
+  // useEffect ensures isLoggedIn only runs on first mount, not re-renders
+  useEffect(() => {
+    if (isLoggedIn()) push('/')
+  }, [])
+
+  const handleSubmit = async () => {
+    if (!user) {
+      // User login
+      const loginResult: LoginPayload = await attemptLogin({ username, password }, loginURL)
+
+      if (!loginResult.success) setIsError(true)
+      else {
+        setIsError(false)
+        setUser(loginResult.user)
+        setJWT(loginResult.JWT)
+        setTemplatePermissions(loginResult.templatePermissions)
+        setOrgList(loginResult?.orgList as OrganisationSimple[])
+      }
+    } else {
+      // Organisation login
+      const { orgId } = orgList[selectedOrgIndex] as OrganisationSimple
+      const authHeader = { Authorization: 'Bearer ' + JWT }
+      const verifyOrgResult = await attemptLogin({ orgId }, loginOrgURL, authHeader)
 
-  const handleSubmit = async (event: any) => {
-    event.preventDefault()
-    const loginResult = await attemptLogin(username, password)
-    if (!loginResult.success) setIsError(true)
-    else {
-      setIsError(false)
-      login(loginResult.JWT)
-      if (history.location?.state?.from) push(history.location.state.from)
-      else push('/')
+      if (verifyOrgResult.success) {
+        finishLogin(verifyOrgResult)
+      }
```

Comment:
Similar thing as suggested before to create utility `attemptOrgLogin` that receives `onSuccess` and `onFailure` and also defines `const loginOrgURL = config.serverREST + '/login-org'` internally.